# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/test/test_mediastream_as_eventarget.html
# Commit: 10d327bb5893
# Full Hash: 10d327bb5893211f030222b02456bfc11e81a943
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1705080
# File Overlap Count: 1
# Description:
#   Bug 1705080, make MediaStreams behave as normal EventTargets, r=jib
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D115269
# ==============================================================================

diff -r 802480d6b25b -r 10d327bb5893 dom/media/test/test_mediastream_as_eventarget.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/media/test/test_mediastream_as_eventarget.html	Tue May 18 16:21:27 2021 +0000
@@ -0,0 +1,33 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>Test MediaStream as EventTarget</title>
+  <script src="/tests/SimpleTest/SimpleTest.js"></script>
+  <link rel="stylesheet" href="/tests/SimpleTest/test.css"/>
+  <script>
+    var ms = new MediaStream();
+    var listenerForFooWasCalled = false;
+    ms.addEventListener("foo", () => { listenerForFooWasCalled = true; });
+    var listenerForAddTrackWasCalled = false;
+    ms.addEventListener("addtrack", () => { listenerForAddTrackWasCalled = true; });
+    var handlerForAddTrackWasCalled = false;
+    ms.onaddtrack = () => { handlerForAddTrackWasCalled = true; };
+
+    ms.dispatchEvent(new Event("foo"));
+    ms.dispatchEvent(new Event("addtrack"));
+
+    ok(listenerForFooWasCalled,
+       "Should have called the event listener for 'foo'");
+    ok(listenerForAddTrackWasCalled,
+       "Should have called the event listener for 'addtrack'");
+    ok(handlerForAddTrackWasCalled,
+       "Should have called the event handler for 'addtrack'")
+  </script>
+</head>
+<body>
+<p id="display"></p>
+<div id="content" style="display: none"></div>
+<pre id="test"></pre>
+</body>
+</html>
